codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -265,18 +271,18 @@ public Http2Stream resetSent() {
         }
 
         @Override
-        public final Object setProperty(Object key, Object value) {
-            return data.put(key, value);
+        public final <K, V> V setProperty(Key<K, V> key, V value) {
+            return properties.add((PropertyKey<K, V>) key, value);
```

Comment:
It will result in `ClassCastException` ... does that not seem reasonable here? Do we prefer `IllegalArgumentException`?